src/test/java/BaseTest.java

```diff
@@ -1,11 +1,357 @@
+    {
+        return new Object[][]
+                {
+                        {"james.patterson@testpro.io", "te$t$tudent"},
+                };
+    }
+
+    protected static void provideEmailCredentials(String email) {
+        //Enters email credentials
+        WebElement emailInput = driver.findElement((By.cssSelector("[type='email']")));
+        emailInput.click();
+        emailInput.clear();
+        emailInput.sendKeys(email);
+    }
+
+    protected static void providePasswordCredentials(String password) {
+        //Enters password credentials
+        WebElement passwordInput = driver.findElement((By.cssSelector("[type='password']")));
+        passwordInput.click();
+        passwordInput.clear();
+        passwordInput.sendKeys(password);
+    }
+
+    protected static void loginButton() {
+        //Clicks login button
+        WebElement loginButton = driver.findElement(By.cssSelector(("button[type='submit']")));
+        loginButton.submit();
+    }
+
+    public static void urlPage()
+    {
+        //Opens web url page
+        String url = "https://qa.koel.app/";
+        driver.get(url);
+    }
+
+    public static void avatarIconVisible() {
+        //Check if the user avatar is displaying
+        WebElement avatar = driver.findElement(By.cssSelector(".avatar"));
+        Assert.assertTrue(avatar.isDisplayed());
+    }
+
+    //Helper functions for Homework17 tasks
+    public void songSearch(String songTitle) throws InterruptedException {
+        //Searches for a song
+        WebElement songSearchInput = driver.findElement(By.cssSelector("#searchForm > input[type=search]"));
+        songSearchInput.click();
+        songSearchInput.clear();
+        songSearchInput.sendKeys("Pluto");
+        Thread.sleep(5000);
+    }
+
+    public void viewAllSongs() throws InterruptedException {
+        //Selects View All button
+        WebElement viewAllButton = driver.findElement(By.cssSelector("#searchExcerptsWrapper > div > div > section.songs > h1 > button"));
+        viewAllButton.click();
+        Thread.sleep(5000);
+    }
+
+    public void firstSongSelection() throws InterruptedException {
+        //Selects first song
+        WebElement firstSong = driver.findElement(By.cssSelector("section#songResultsWrapper tr.song-item td.title"));
+        firstSong.click();
+        Thread.sleep(5000);
+    }
+
+    public void addToButton() throws InterruptedException {
+        //Clicks AddTo Button
+        WebElement addToButton = driver.findElement(By.cssSelector("button.btn-add-to"));
+        addToButton.click();
+        Thread.sleep(5000);
+    }
+
+    public void selectedSongAddedToPlaylist() throws InterruptedException {
+        //Adds selected song to playlist
+        //Thread.sleep(3000);
+        WebElement songPlaylist = driver.findElement(By.xpath("/html/body/div[1]/div/div[1]/section[1]/section[11]/header/div[3]/div/section[1]/ul/li[5]"));
+        songPlaylist.click();
+        Thread.sleep(20000);
```

Comment:
Thread. sleep is bad! It blocks the current thread and renders it unusable for further work.
Using Thread. sleep() frequently in an automation framework is not a good practice.

Please use the Explicit Wait command, the WebDriver is directed to wait until a certain condition occurs before proceeding with executing the code.